src/app/Services/api.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { BalanceDto } from '../balanceDto';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ApiService {
+  id = 6;
+  clockInUrl = 'https://localhost:44385/api/ClockInOut/ClockIn/';
+  clockOutUrl = 'https://localhost:44385/api/ClockInOut/ClockOut/';
+  balanceUrl = 'https://localhost:44385/api/ClockInOut/Balance/';
+  BalanceToThisDayUrl =
+    'https://localhost:44385/api/ClockInOut/BalanceToThisDay/';
```

Comment:
we wszystkich czterech przypadkach powtarzamy kod https://localhost:44385/api. można ten fragment wrzucić do osobnego pola np. apiBaseUrl = https://localhost:44385/api/ a następnie zaimplementować ten fragment w pozostałych linkach.

```suggestion
  clockInUrl = `${this.apiBaseUrl}/ClockInOut/ClockIn/`;
  clockOutUrl = `${this.apiBaseUrl}/ClockInOut/ClockOut/`;
  balanceUrl = `${this.apiBaseUrl}/ClockInOut/Balance/`;
  BalanceToThisDayUrl =
    `${this.apiBaseUrl}/api/ClockInOut/BalanceToThisDay/`;
```
Jeśli mamy fragmenty kodu, które nam się powtarzają to starajmy się je zawsze opakowywać w reużywalne pola/metody. To ułatwia utrzymywanie aplikacji. Na przykładzie tego kodu - jeśli zechcemy zmienić localhost:44385 na jakiś inny port lub całkowicie inny adres to musielibyśmy zrobić to w czterech miejscach, a kiedy zapiszemy tą wartość w polu - wystarczy dokonać zmiany w jednym miejscu.